R/functions_analysis.R

```diff
@@ -0,0 +1,55 @@
+#' Calculate enrichment of cells per sample per cluster.
+#' 
+#' @param sc Seurat object.
+#' @return A table with counts, odd ratios and p-values.
+cells_fisher = function(sc) {
+  cell_samples = sc[[]] %>% dplyr::pull(orig.ident) %>% unique() %>% sort()
+  cell_clusters = sc[[]] %>% dplyr::pull(seurat_clusters) %>% unique() %>% sort()
+  out = matrix(0+NA, nrow=0, ncol=length(cell_clusters)) %>% as.data.frame()
+  for(s in cell_samples) {
+    ft.list = lapply(cell_clusters, function(cl) { 
+      a = sc[[]] %>% dplyr::filter(orig.ident==s, seurat_clusters==cl) %>% dplyr::count() %>% as.numeric()
+      b = sc[[]] %>% dplyr::filter(orig.ident!=s, seurat_clusters==cl) %>% dplyr::count() %>% as.numeric()
+      c = sc[[]] %>% dplyr::filter(orig.ident==s, seurat_clusters!=cl) %>% dplyr::count() %>% as.numeric()
+      d = sc[[]] %>% dplyr::filter(orig.ident!=s, seurat_clusters!=cl) %>% dplyr::count() %>% as.numeric()
+      tbl.2by2 = matrix(c(a, b, c, d), ncol=2, nrow=2, byrow=TRUE)
+      ft = fisher.test(tbl.2by2, alternative="greater")
```

Comment:
Mmmh ... I think in this very case, we want to know, cells from which sample form which cluster, and under-representation would be confusing. If we have 2 samples and a cluster, sample 1 might be under-respresented, and automatically, sample 2 over-represented, so both are significant. But what we care for is to know that cells from sample 2 make up that cluster.

Reply:
A practical example would be: Sample A, B and C can be grouped together in 15 clusters (==cell types). Now, it may be important to know whether a certain cell type is missing/reduced in sample A. I do not think that automatically the other samples are then overrepresented.

Reply:
I think it is a little confusing for this table, but I'll change it for you. I don't mind enough to argue =).

Reply:
Thats nice :-)